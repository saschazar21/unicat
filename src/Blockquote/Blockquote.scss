@import '../__styles__/variables';

.wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  font-size: fontSize(h5);
  margin: spacing() spacing(xl);
}

.author {
  align-self: flex-end;
  color: color(grey, false, $systemColors);
  font-size: fontSize(small);
  font-style: italic;
  text-transform: uppercase;
  letter-spacing: .25em;
  margin-top: spacing();
}

.quote {
  font-weight: bold;
  text-align: center;
  
  &::before,
  &::after {
    color: color(grey, false, $systemColors);
    font-size: fontSize(small);
    font-weight: normal;
  }
  
  &::before {
    content: '<<';
    margin-right: spacing(xs);
  }
  
  &::after {
    content: '>>';
    margin-left: spacing(xs);
  }
}
