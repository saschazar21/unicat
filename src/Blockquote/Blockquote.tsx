import React, { Component, ReactNode } from 'react';
import classnames from 'classnames';

import styles from './Blockquote.scss';

export interface BlockquoteProps {
  author?: string;
  children: ReactNode;
  className?: string;
  cite?: string;
}

export default class Blockquote extends Component<BlockquoteProps> {
  public render() {
    const {
      author,
      children,
      className: customClassName,
      ...props
    } = this.props;

    const className = classnames(styles.wrapper, customClassName);

    return (
      <blockquote className={className} {...props}>
        <span className={styles.quote}>{children}</span>
        {author && <span className={styles.author}>â€• {author}</span>}
      </blockquote>
    );
  }
}
