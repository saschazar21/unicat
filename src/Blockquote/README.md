Renders a quote, with an optional `author` reference

```jsx
<Blockquote
  cite="https://www.goodreads.com/quotes/tag/day"
  author="Edgar Allan Poe"
>
  Those who dream by day are cognizant of many things which escape those who
  dream only by night.
</Blockquote>
```
