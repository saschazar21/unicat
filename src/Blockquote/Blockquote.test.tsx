import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Blockquote from './Blockquote';

describe('Blockquote', () => {
  it('renders', () => {
    const wrapper = shallow(<Blockquote>A test quote</Blockquote>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders an author', () => {
    const wrapper = shallow(<Blockquote author="Test">Test quote</Blockquote>);

    expect(wrapper.find('.author')).toHaveLength(1);
  });
});
