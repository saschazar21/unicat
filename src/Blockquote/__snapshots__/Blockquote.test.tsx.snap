// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote renders 1`] = `
<blockquote
  className="wrapper"
>
  <span
    className="quote"
  >
    A test quote
  </span>
</blockquote>
`;
